docs/commands.md

```diff
@@ -0,0 +1,270 @@
+  They should not appear in "finished" files.
+  Transient commands typically begin with the symbol `#`.
+  `#check` is a standard example of a transient command.
+
+* *Permanent* commands modify the environment. Removing a permanent command from a file may affect
+  the following proofs. `set_option class.instance_max_depth 500` is a standard example of a
+  permanent command.
+
+User-defined commands can have unintuitive interactions with the parser. For the most part, this is
+not something to worry about. However, you may occasionally see strange error messages when using
+mathlib commands: for instance, running these commands immediately after `import file.name` will
+produce an error. An easy solution is to run a built-in no-op command in between, for example:
+
+```
+import data.nat.basic
+
+run_cmd tactic.skip -- this serves as a "barrier" between `import` and `#find`
+
+#find _ + _ = _ + _
+```
+
+## find
+
+The `find` command from `tactic.find` allows to find lemmas using
+pattern matching. For instance:
+
+```lean
+import tactic.find
+
+#find _ + _ = _ + _
+#find (_ : ℕ) + _ = _ + _
+```
+
+## Localized notation
+
+This consists of two user-commands which allow you to declare notation and commands localized to a namespace.
+
+* Declare notation which is localized to a namespace using:
+```
+localized "infix ` ⊹ `:60 := my_add" in my.add
+```
+* After this command it will be available in the same section/namespace/file, just as if you wrote `local infix ` ⊹ `:60 := my_add`
+* You can open it in other places. The following command will declare the notation again as local notation in that section/namespace/files:
+```
+open_locale my.add
+```
+* More generally, the following will declare all localized notation in the specified namespaces.
+```
+open_locale namespace1 namespace2 ...
+```
+* You can also declare other localized commands, like local attributes
+```
+localized "attribute [simp] le_refl" in le
+```
+* Warning 1: as a limitation on user commands, you cannot put `open_locale` directly after your imports. You have to write another command first (e.g. `open`, `namespace`, `universe variables`, `noncomputable theory`, `run_cmd tactic.skip`, ...).
+* Warning 2: You have to fully specify the names used in localized notation, so that the localized notation also works when the appropriate namespaces are not opened.
+
+## reassoc_axiom
+
+When declaring a class of categories, the axioms can be reformulated to be more amenable
+to manipulation in right associated expressions:
+
+```lean
+class some_class (C : Type) [category C] :=
+(foo : Π X : C, X ⟶ X)
+(bar : ∀ {X Y : C} (f : X ⟶ Y), foo X ≫ f = f ≫ foo Y)
+
+reassoc_axiom some_class.bar
+```
+
+The above will produce:
+
+```lean
+lemma some_class.bar_assoc {Z : C} (g : Y ⟶ Z) :
+  foo X ≫ f ≫ g = f ≫ foo Y ≫ g := ...
+```
+
+Here too, the `reassoc` attribute can be used instead. It works well when combined with
```

Comment:
What is this "too" referring to?

Reply:
I don't know. It's copied from the entry in `tactics.md`.